src/shared/activationReloadState.ts

```diff
@@ -22,15 +22,11 @@ export interface SamInitState {
  */
 export class ActivationReloadState {
     public getSamInitState(): SamInitState | undefined {
-        const activationPath = this.extensionContext.globalState.get<string>(ACTIVATION_LAUNCH_PATH_KEY)
-
-        return activationPath
-            ? {
-                  path: activationPath,
-                  runtime: this.extensionContext.globalState.get<string>(SAM_INIT_RUNTIME_KEY),
-                  isImage: this.extensionContext.globalState.get<boolean>(SAM_INIT_IMAGE_BOOLEAN_KEY),
-              }
-            : undefined
+        return {
+            path: this.extensionContext.globalState.get<string>(ACTIVATION_LAUNCH_PATH_KEY),
+            runtime: this.extensionContext.globalState.get<string>(SAM_INIT_RUNTIME_KEY),
+            isImage: this.extensionContext.globalState.get<boolean>(SAM_INIT_IMAGE_BOOLEAN_KEY),
```

Comment:
Don't need to check the implicit "are we reloading" flag, instead callers must check `ext.didReload()`. That is easier to reason about and avoids needing to understand the order-of-operations that would result in `getSamInitState()` returning `undefined`.